Code/src/main/java/nl/utwente/viskell/ui/components/FunApplyBlock.java

```diff
@@ -0,0 +1,224 @@
+            dragContext.setDragInitAction(c -> {this.curried = false;});
+            dragContext.setDragFinishAction(c -> {
+                double height = this.inputType.getHeight();
+                this.curried = this.inputType.getLayoutY() > height;
+                this.inputType.relocate(0, this.curried ? 2*height : 0);
+                FunApplyBlock.this.initiateConnectionChanges();
+            });
+        }
+
+        @Override
+        public double computePrefHeight(double width) {
+            double height = this.inputType.prefHeight(width)*2;
+            this.dragContext.setDragLimits(new BoundingBox(0, 0, 0, height));
+            return height;
+        }
+
+        /** Shift related things down to some distance. */
+        private void dragShift(double y) {
+            double height = this.inputType.getHeight();
+            this.anchor.setLayoutY(y);
+            this.anchor.setOpacity(1 - (y/(height*1.5)));
+            this.anchor.setVisible(y < height);
+            FunApplyBlock.this.dragShiftOuput(y - height);
+        }
+
+    }
+    
+    /** The information about the function. */
+    private FunctionInfo funInfo;
+
+    private List<FunInputAnchor> inputs;
+    
+    /** Text label for the output type */
+    private final Label resTypeLabel;
+    
+    /** The uncurried output type */
+    private Type resType;
+    
+    /** The result anchor of this function. */
+    private final OutputAnchor output;
+
+    /** The space containing anchors and type labels. */
+    @FXML private Pane bodySpace;
+    
+    /** The Label in which the information of the function is displayed. */
+    @FXML private Label functionInfo;
+    
+    public FunApplyBlock(FunctionInfo funInfo, CustomUIPane pane) {
+        super(pane);
+        this.loadFXML("FunApplyBlock");
+
+        this.funInfo = funInfo;
+        this.functionInfo.setText(funInfo.getName());
+        this.inputs = new ArrayList<>();
+        this.output = new OutputAnchor(this, new Binder("res"));
+        
+        this.resTypeLabel = new Label("");
+        this.resTypeLabel.setMinWidth(USE_PREF_SIZE);
+        this.resTypeLabel.getStyleClass().add("resultType");
+        VBox outputSpace = new VBox(this.resTypeLabel, this.output);
+        outputSpace.setAlignment(Pos.CENTER);
+        
+        Type t = funInfo.getFreshSignature();
+        while (t instanceof FunType) {
+            FunType ft = (FunType) t;
+            FunInputAnchor ia = new FunInputAnchor();
+            this.inputs.add(ia);
+            t = ft.getResult();
+        }
+
+        Pane inputSpace = new HBox(10, this.inputs.toArray(new Node[this.inputs.size()]));
+        this.bodySpace.getChildren().addAll(inputSpace, outputSpace);
+        outputSpace.layoutYProperty().bind(inputSpace.heightProperty());
+        outputSpace.layoutXProperty().bind(inputSpace.widthProperty().divide(2).subtract(resTypeLabel.widthProperty().divide(2)));
```

Comment:
It is simply centering one thing to center of another thing.
I searched for hours for better solution with standard layout blocks, but didn't find any.
The problem is with how event handling is coupled with node/parent hierachy, and that doesn't work well with moving parts.
And workarounds with setMouseTransparent don't work either, because that also affect all (even indirect) children nodes.